src/rendering/RenderUtil.cc

```diff
@@ -1810,6 +1812,17 @@ void RenderUtilPrivate::CreateEntitiesFirstUpdate(
 
   if (this->enableSensors)
   {
+    // Create custom sensors
+    _ecm.Each<components::CustomSensor, components::ParentEntity>(
+      [&](const Entity &_entity,
+          const components::CustomSensor *_customSensor,
+          const components::ParentEntity *_parent)->bool
+        {
+          this->AddNewSensor(_ecm, _entity, _customSensor->Data(),
```

Comment:
Do you mean this section will be triggered even if add say an odometer custom sensor (https://github.com/ignitionrobotics/ign-sensors/tree/main/examples/custom_sensor) which is not a custom "rendering" sensor ?

Reply:
Yes, I believe this will be triggered for other custom sensors like the odometer example. If you see
https://github.com/ignitionrobotics/ign-gazebo/blob/9cd15fc941b0e9435abb6fa3598b54527009a074/src/SdfEntityCreator.cc#L937-L942

The `CustomSensor` component is added for any sensor with `type=custom` whether it's a rendering sensor or not. This  `Each` call in `RenderUtil.cc` iterates through all entities that have the `CustomSensor` component.

Reply:
Hmm not sure how to resolve that.

Reply:
maybe add `ignition:type="your_custom_sensor_type"` to the sdf so that the  custom sensor system will ignore any custom sensor types that it does not know and no operation will be performed.

Another thing I noticed is that the create sensor callback is overriden. So it looks like the `sensors` system can not be used at the same time as any custom rendering sensor systems? Something to think about later is to allow custom systems to register additional CreateSensor callbacks instead of having just one callback

Reply:
As of now : The custom rendering system + sensor cannot be used at the same time as the original ``ignition-gazebo-sensors-system`` plus a normal ``camera``.